python/semantic_kernel/planners/function_calling_stepwise_planner/function_calling_stepwise_planner.py

```diff
@@ -0,0 +1,249 @@
+# Copyright (c) Microsoft. All rights reserved.
+
+import asyncio
+import json
+import logging
+import os
+from copy import copy
+from typing import Optional
+
+import yaml
+
+from semantic_kernel.connectors.ai.chat_completion_client_base import (
+    ChatCompletionClientBase,
+)
+from semantic_kernel.connectors.ai.open_ai.contents.open_ai_chat_message_content import OpenAIChatMessageContent
+from semantic_kernel.connectors.ai.open_ai.prompt_execution_settings.open_ai_prompt_execution_settings import (
+    OpenAIChatPromptExecutionSettings,
+)
+from semantic_kernel.connectors.ai.open_ai.utils import (
+    get_tool_call_object,
+)
+from semantic_kernel.contents.chat_history import ChatHistory
+from semantic_kernel.exceptions.planner_exceptions import PlannerInvalidConfigurationError
+from semantic_kernel.functions.kernel_arguments import KernelArguments
+from semantic_kernel.functions.kernel_function import KernelFunction
+from semantic_kernel.kernel import Kernel
+from semantic_kernel.kernel_pydantic import KernelBaseModel
+from semantic_kernel.planners.function_calling_stepwise_planner.function_calling_stepwise_planner_options import (
+    FunctionCallingStepwisePlannerOptions,
+)
+from semantic_kernel.planners.function_calling_stepwise_planner.function_calling_stepwise_planner_result import (
+    FunctionCallingStepwisePlannerResult,
+    UserInteraction,
+)
+from semantic_kernel.planners.planner_extensions import PlannerKernelExtension
+from semantic_kernel.prompt_template.kernel_prompt_template import KernelPromptTemplate
+from semantic_kernel.prompt_template.prompt_template_config import PromptTemplateConfig
+
+CUR_DIR = os.path.dirname(os.path.realpath(__file__))
+PLAN_YAML_FILE_PATH = os.path.join(CUR_DIR, "generate_plan.yaml")
+STEP_PROMPT_FILE_PATH = os.path.join(CUR_DIR, "step_prompt.txt")
+
+STEPWISE_PLANNER_PLUGIN_NAME = "StepwisePlanner_Excluded"
+
+STEPWISE_USER_MESSAGE = (
+    "Perform the next step of the plan if there is more work to do."
+    "When you have reached a final answer, use the UserInteraction-SendFinalAnswer"
```

Comment:
should the function name be in there as a param? so that we only have to change it once?

Reply:
We're not going through the kernel_function invoke internal path, so there's no way that the user message is getting parsed before sending it to the LLM. Not sure what advantage we'd gain by introducing the function name as a param in this user message...